common/conf.hpp

```diff
@@ -273,8 +275,14 @@ class MenderConfig : public cfg_parser::MenderConfigFromFile {
 		vector<string>::const_iterator end,
 		const CliApp &app);
 
+	const http::ClientConfig &GetHttpClientConfig() const {
```

Comment:
Q: why `const http::ClientConfig` ? Did we not have a conversation about this not making a difference before? :thinking:

Reply:
Also, good idea!

Reply:
Yes, but it's a reference, not a value, and then it makes a difference.

Using `const &` is actually a micro-optimization, and not really critical. My intention here was to return plain `http::ClientConfig`, no const, no reference. However, because it happens to be a member of the class, and we don't need to construct a new one, we can return a const reference as an alternative. As long as the caller assigns it to a new struct, this makes no practical difference, it just saves a few CPU cycles.

The only case where it would make a difference is if the caller assigns it to a reference. If we later changed it to return plain `http::ClientConfig`, which we would have to if the function needs to construct the data, then this wouldn't work anymore.

So yeah, I could change to plain `http::ClientConfig` since it is, in an API sense, more correct. I'm fine with either.

Reply:
Alright, makes sense. Also fine with either